File: app/src/main/java/com/levibostian/androidblanky/module/ManagerModule.java
package com.levibostian.androidblanky.module;

import android.content.Context;
import android.content.SharedPreferences;

import com.levibostian.androidblanky.manager.UserManager;
import com.levibostian.androidblanky.manager.api.ReposApiManager;
import com.levibostian.androidblanky.service.GitHubApi;

import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;

@Module
public class ManagerModule {

    private Context mContext;

    public ManagerModule(Context context) {
        mContext = context;
    }

    @Provides
    public UserManager provideUserManager(SharedPreferences sharedPreferences) {
        return new UserManager(mContext, sharedPreferences);
    }

    @Provides
    public ReposApiManager provideReposApiManager(GitHubApi api, Retrofit retrofit) {
        return new ReposApiManager(api, retrofit);
    }

}

File: app/src/main/java/com/levibostian/androidblanky/module/ManagerModule.kt
package com.levibostian.androidblanky.module

import android.accounts.AccountManager
import com.levibostian.androidblanky.service.manager.AppDeviceAccountManager
import com.levibostian.androidblanky.service.manager.DeviceAccountManager
import com.levibostian.androidblanky.service.manager.NotificationChannelManager
import com.levibostian.androidblanky.service.manager.UserManager
import org.koin.android.ext.koin.androidContext
import org.koin.core.module.Module
import org.koin.dsl.module

object ManagerModule {

    fun get(): Module {
        return module {
            factory { AppDeviceAccountManager(get()) as DeviceAccountManager }
            factory { UserManager(get(), get()) }
            factory { NotificationChannelManager(androidContext()) }
        }
    }

}